### ISSUE TITLE
Game gets stuck on a new campaign.

### ISSUE THREAD

**Issue by digitalinferno** [GitoMat/Shogun2-Linux-Fix#2]
Everything works fine (historical battles, custom battles, settings, etc.), but when starting a new campaign, the game gets stuck after the introduction video. No errors from the terminal, but the CPU usage spikes.

**Comment by GitoMat** [comment]
For me starting a campaign works flawlessly on Ubuntu 24.04.1 - what distro are you using? And what makes you think this problem related to these patches?

**Comment by digitalinferno** [comment]
Maybe something has changed and the patches need an update... I asked you just to rule out a possible cause. I just finished installing Fedora 41 and the problem persists, so there's probably something wrong with mesa (or another library), I don't know. Whether I use Intel Igp or AMD GPU drivers, nothing changes. If I remember correctly the last version of Fedora working for me is 38.

**Comment by GitoMat** [comment]
I now tried it on Ubuntu 24.10 which has glibc version 2.40 and it works. Fedora 41 has the same glibc verison, so as far as I can say the problem is most likely not with the glibc patches in this repo.

**Comment by digitalinferno** [comment]
Thank you for your time; I truly appreciate it.

**Comment by GitoMat** [comment]
You are welcome. What you could try is to lauch Steam from the terminal, then you might be able to see the actual problem in the console output of the game. This at least has helped me in the past ...

**Comment by digitalinferno** [comment]
I just tried, but the console doesn't provide any useful output. The strange thing is that the game itself works fine, except that the campaign won't start. I used gdb and valgrind to debug the issue, and it seems to be a case of memory corruption. However, the backtrace can't access the memory address. This might be related to a bug in [libsteam_api.so](https://github.com/ValveSoftware/steam-for-linux/issues/8542#top). On my system, if the username directory is too long, some games fails to start.
